File: DP/70. Climbing Stairs.java
class Solution {
    HashMap<Integer,Integer> x = new HashMap<Integer,Integer>();
    public int climbStairs(int n) {
        if(n<4){
            return n;
        }else if(!x.containsKey(n)){
            x.put(n,climbStairs(n-1) + climbStairs(n-2));
        }
        return x.get(n);
    }
}

File: DP/70. Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n==1:
            return 1
        p1 = 1
        p2 = 2
        for cnt in range(2,n):
            tmp = p1
            p1 = p2
            p2 = tmp + p1
        return p2